Code/Framework/AzFramework/AzFramework/Spawnable/SpawnableEntitiesManager.cpp

```diff
@@ -482,8 +550,10 @@ namespace AzFramework
             {
                 if (entity != nullptr)
                 {
+                    // Setting it to 0 is needed to avoid the infite loop between GameEntityContext and SpawnableEntitiesManager.
+                    entity->SetSpawnTicketId(0);
                     GameEntityContextRequestBus::Broadcast(
-                        &GameEntityContextRequestBus::Events::DestroyGameEntityAndDescendants, entity->GetId());
+                        &GameEntityContextRequestBus::Events::DestroyGameEntity, entity->GetId());
```

Comment:
That would delay the entity setting the spawn ticket id to 0 until it can find the handler. I don't think that's the intended pattern here. The SpawnableEntitiesManager should be responsible for managing the ticket inside the entity and the GameEntityContext should be responsible for destroying it. Besides, if the plans to use AzFramework::Scene instead of GameEntityContextRequestBus we would get much more significant performance gains than using the above suggested optimization.
So I suggest that we keep this in mind and revisit the optimization later if we cannot use AzFramework::Scene.

Reply:
"That would delay the entity setting the spawn ticket id to 0 until it can find the handler." Can you elaborate as to why this is a problem? This is just a bus hygiene best practice to avoid multiple lookups to the handler, all `SetSpawnTicketId` calls made subsequent to the first entity would also be delayed waiting on the bus invocation, this just prevents us from looking up the bus handler N times, I don't think it's a separation of concerns issue at all. This isn't a needed cleanup, but it is a pattern I'd like to see us promote where possible to avoid unnecessary pessimizations when using the bus system.

Reply:
Had to discuss this with @NicholasVanSickle because I wasn't aware that a broadcast with a lambda was even an option. With the Spawnable Entities Interface/Manager preferring batches, the suggested broadcast would actually help remove some of the overhead of the EBus from the loops, so this is a suggestion that should be applied throughout the Spawnable Entitities Manager. I agree with @srikappa-amzn though that the move to AzFramework::Scene, which will remove the EBus entirely, will be even better.

Reply:
Okay. Using Lambdas with ebuses seems like a straight forward change that should improve performance. There are other places in SpawnableEntitiesManager where we make ebus calls too and we should change those also then. I'll create a separate task for this though to get the despawning out of the door first. If possible, I'd also like to add benchmarks around SEM and see the performance gains for the switch to use lambdas with ebuses. That would be fun!

Reply:
👍 Sounds good to me!